analysis/tests/src/not_compiled/expected/Diagnostics.res.txt

```diff
@@ -0,0 +1,13 @@
+[{
```

Comment:
There's the question of what to do with parser recovery, see if this is too noisy. Always possible to report only the first thing.
Something to evaluate after hooking this command to the editor in `server.ts`.

Reply:
Can you elaborate a little more? It is related to this PR?

Reply:
So the parser runs in recovery mode, and keeps on parsing after the first syntax error.
You can see this in your test where 3 things are reported for the same issue.
Now, one could just stop after the first report. Which would take care of the noise in this example.
However, it would be nice to report more than one (syntax) error at once, if possible, when these errors happen in different regions.
Since this PR exposes the syntax errors to users, it's part of this PR to evaluate how those errors look like in practice.
Then based on that, decide how to proceed.

Reply:
It might just be OK the way it is now. But one would need to experiment once the command it hooked up to the extension.

Reply:
Ok, I'll test when #453 is finished because @zth will update the vscode client.

Reply:
Another related aspect, that also needs to wait, is about making sure there is no double reporting when the compiler is already reporting the syntax error.

Reply:
E.g.:
1 introduce syntax error
2 compile
3 edit unrelated part of the file

Then you get a file that is unsaved, with syntax errors, but where the syntax error was reported already.